Specs/Scene/ModelExperimental/SkinningPipelineStageSpec.js

```diff
@@ -0,0 +1,183 @@
+import {
+  combine,
+  GltfLoader,
+  Matrix4,
+  ModelExperimentalType,
+  Resource,
+  ResourceCache,
+  ShaderBuilder,
+  _shadersSkinningStageVS,
+  SkinningPipelineStage,
+} from "../../../Source/Cesium.js";
+import createScene from "../../createScene.js";
+import waitForLoaderProcess from "../../waitForLoaderProcess.js";
+import ShaderBuilderTester from "../../ShaderBuilderTester.js";
+
+describe(
+  "Scene/ModelExperimental/SkinningPipelineStage",
+  function () {
+    let scene;
+    const gltfLoaders = [];
+
+    beforeAll(function () {
+      scene = createScene();
+    });
+
+    afterAll(function () {
+      scene.destroyForSpecs();
+    });
+
+    afterEach(function () {
+      const gltfLoadersLength = gltfLoaders.length;
+      for (let i = 0; i < gltfLoadersLength; ++i) {
+        const gltfLoader = gltfLoaders[i];
+        if (!gltfLoader.isDestroyed()) {
+          gltfLoader.destroy();
+        }
+      }
+      gltfLoaders.length = 0;
+      ResourceCache.clearForSpecs();
+    });
+
+    function getOptions(gltfPath, options) {
+      const resource = new Resource({
+        url: gltfPath,
+      });
+
+      return combine(options, {
+        gltfResource: resource,
+        incrementallyLoadTextures: false, // Default to false if not supplied
+      });
+    }
+
+    function loadGltf(gltfPath, options) {
+      const gltfLoader = new GltfLoader(getOptions(gltfPath, options));
+      gltfLoaders.push(gltfLoader);
+      gltfLoader.load();
+
+      return waitForLoaderProcess(gltfLoader, scene);
+    }
+
+    const simpleSkinUrl =
+      "./Data/Models/GltfLoader/SimpleSkin/glTF/SimpleSkin.gltf";
+    const cesiumManUrl =
+      "./Data/Models/DracoCompression/CesiumMan/CesiumMan.gltf";
+
+    it("processes skin with two joints", function () {
+      const mockJointMatrices = [
+        new Matrix4(),
+        Matrix4.clone(Matrix4.IDENTITY),
+      ];
+      const renderResources = {
+        attributes: [],
+        shaderBuilder: new ShaderBuilder(),
+        attributeIndex: 1,
+        model: {
+          type: ModelExperimentalType.TILE_GLTF,
+        },
+        runtimeNode: {
+          computedJointMatrices: mockJointMatrices,
+        },
+      };
```

Comment:
If we add a third test for a model with >4 joints, consider wrapping this setup code in a function. Our mocking code tends to get messy. See `mockRenderResources` in `FeatureIdPipelineStage`